app/src/main/java/org/p2p/wallet/auth/gateway/repository/mapper/ChaCha20Poly1305Encryptor.kt

```diff
@@ -0,0 +1,37 @@
+package org.p2p.wallet.auth.gateway.repository.mapper
+
+import org.bouncycastle.crypto.modes.ChaCha20Poly1305
+import org.bouncycastle.crypto.params.KeyParameter
+import org.bouncycastle.crypto.params.ParametersWithIV
+import org.p2p.wallet.utils.processBytesKt
+
+class ChaCha20Poly1305Encryptor(
+    private val chaCha20Poly1305: ChaCha20Poly1305
+) {
+    fun encryptData(
+        privateKey: ByteArray,
+        nonce: ByteArray,
+        dataToEncrypt: ByteArray
+    ): ByteArray {
+        chaCha20Poly1305.init(
+            true,
+            ParametersWithIV(
+                KeyParameter(privateKey.copyOf()),
+                nonce.copyOf()
+            )
+        )
+
+        val resultSize = chaCha20Poly1305.getOutputSize(dataToEncrypt.size)
+        val encryptionResult = ByteArray(resultSize)
+
+        val finalOff = chaCha20Poly1305.processBytesKt(
+            inBytes = dataToEncrypt,
+            len = dataToEncrypt.size,
+            outBytes = encryptionResult,
+        )
+        chaCha20Poly1305.doFinal(encryptionResult, finalOff)
+
+        chaCha20Poly1305.reset()
+        return encryptionResult
```

Comment:
@gslevinkov can you ask somebody from management if it's okay that we show what type of encryption we do here?
Since it's an open-source, it may be dangerous, but I am not sure